README.md
# react-style-reset

[![GitHub Actions](https://github.com/iiroj/react-style-reset/workflows/Tags/badge.svg)](https://github.com/iiroj/react-style-reset/actions)
[![version](https://img.shields.io/npm/v/react-style-reset.svg)](https://www.npmjs.com/package/react-style-reset)
[![code size](https://img.shields.io/github/languages/code-size/iiroj/react-style-reset.svg)](https://github.com/iiroj/react-style-reset)
[![dependencies](https://img.shields.io/david/iiroj/react-style-reset.svg)](https://github.com/iiroj/react-style-reset/blob/master/package.json)
[![devDependencies](https://img.shields.io/david/dev/iiroj/react-style-reset.svg)](https://github.com/iiroj/react-style-reset/blob/master/package.json)

An object-style CSS Reset for React.

## Features

This module includes [Eric Meyer's Reset CSS 2.0](https://meyerweb.com/eric/tools/css/reset/) written as "object styles", to be easily used with [styled-component](https://www.styled-components.com/) and [emotion](https://emotion.sh/). Also included is a template literal version.

Also included is the popular [Normalize.css](https://necolas.github.io/normalize.css/), a modern HTML5-ready alternative to CSS resets, in both object and template literal versions.

## Usage

## Reset.css

With object styles:

```js
import reset from 'react-style-reset';
import { injectGlobal } from 'emotion';

injectGlobal(reset, {
  ...
});
```

With template literal styles:

```js
import reset from 'react-style-reset/string';
import { createGlobalStyle } from 'styled-components';

const GlobalStyles = createGlobalStyle`
  ${reset};

  ...
`;
```

## Normalize.css

With object styles:

```js
import { normalize } from 'react-style-reset';
import { injectGlobal } from 'emotion';

injectGlobal(normalize, {
  ...
});
```

With template literal styles:

```js
import { normalize } from 'react-style-reset/string';
import { createGlobalStyle } from 'styled-components';

const GlobalStyles = createGlobalStyle`
  ${normalize};

  ...
`;
```

## Credits

Eric Meyer http://meyerweb.com/eric/tools/css/reset/
License: none (public domain)

normalize.css v8.0.0 | MIT License | github.com/necolas/normalize.css

## License

MIT — see [LICENCE](LICENSE)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [1.3.2](https://github.com/iiroj/react-style-reset/compare/v1.3.1...v1.3.2) (2020-08-20)


### Bug Fixes

* use correct border-box instead of borderBox ([67fe9b0](https://github.com/iiroj/react-style-reset/commit/67fe9b0951bb899a3462659934c585dbb18ae0e9))

### [1.3.1](https://github.com/iiroj/react-style-reset/compare/v1.3.0...v1.3.1) (2020-03-14)

## [1.3.0](https://github.com/iiroj/react-style-reset/compare/v1.2.15...v1.3.0) (2019-12-30)


### Features

* use mjs extension for esm module ([6d91a99](https://github.com/iiroj/react-style-reset/commit/6d91a99c21ef04f32bdbabbd13ba3daf8b87137b))

### [1.2.15](https://github.com/iiroj/react-style-reset/compare/v1.2.14...v1.2.15) (2019-12-01)

### [1.2.14](https://github.com/iiroj/react-style-reset/compare/v1.2.13...v1.2.14) (2019-09-25)

### [1.2.13](https://github.com/iiroj/react-style-reset/compare/v1.2.12...v1.2.13) (2019-09-22)

### [1.2.12](https://github.com/iiroj/react-style-reset/compare/v1.2.11...v1.2.12) (2019-08-27)

### [1.2.11](https://github.com/iiroj/react-style-reset/compare/v1.2.10...v1.2.11) (2019-07-29)



### [1.2.10](https://github.com/iiroj/react-style-reset/compare/v1.2.9...v1.2.10) (2019-07-11)



### [1.2.9](https://github.com/iiroj/react-style-reset/compare/v1.2.8...v1.2.9) (2019-07-07)


### Build System

* add husky, commitlint and lint-staged ([a9e1ea8](https://github.com/iiroj/react-style-reset/commit/a9e1ea8))



### [1.2.8](https://github.com/iiroj/react-style-reset/compare/v1.2.7...v1.2.8) (2019-07-06)



### [1.2.7](https://github.com/iiroj/react-style-reset/compare/v1.2.6...v1.2.7) (2019-06-02)



### [1.2.6](https://github.com/iiroj/react-style-reset/compare/v1.2.5...v1.2.6) (2019-05-13)



## [1.2.5](https://github.com/iiroj/react-style-reset/compare/v1.2.4...v1.2.5) (2019-04-28)



## [1.2.4](https://github.com/iiroj/react-style-reset/compare/v1.2.3...v1.2.4) (2019-03-27)



## [1.2.3](https://github.com/iiroj/react-style-reset/compare/v1.2.2...v1.2.3) (2019-03-11)



## [1.2.2](https://github.com/iiroj/react-style-reset/compare/v1.2.1...v1.2.2) (2019-03-02)



## [1.2.1](https://github.com/iiroj/react-style-reset/compare/v1.2.0...v1.2.1) (2019-03-02)



# [1.2.0](https://github.com/iiroj/react-style-reset/compare/v1.1.9...v1.2.0) (2019-02-23)


### Features

* use eslint instead of tslint ([70caa51](https://github.com/iiroj/react-style-reset/commit/70caa51))



<a name="1.1.9"></a>
## [1.1.9](https://github.com/iiroj/react-style-reset/compare/v1.1.8...v1.1.9) (2019-02-05)



<a name="1.1.8"></a>
## [1.1.8](https://github.com/iiroj/react-style-reset/compare/v1.1.7...v1.1.8) (2019-02-02)


### Bug Fixes

* table borderCollapse as string literal so solve ts error when injecting ([3eae980](https://github.com/iiroj/react-style-reset/commit/3eae980))
* use named exports in Rollup ([5963072](https://github.com/iiroj/react-style-reset/commit/5963072))



<a name="1.1.7"></a>
## [1.1.7](https://github.com/iiroj/react-style-reset/compare/v1.1.6...v1.1.7) (2019-01-26)



<a name="1.1.6"></a>
## [1.1.6](https://github.com/iiroj/react-style-reset/compare/v1.1.5...v1.1.6) (2019-01-26)



<a name="1.1.5"></a>
## [1.1.5](https://github.com/iiroj/react-style-reset/compare/v1.1.4...v1.1.5) (2019-01-26)



<a name="1.1.4"></a>
## [1.1.4](https://github.com/iiroj/react-style-reset/compare/v1.1.3...v1.1.4) (2019-01-15)



<a name="1.1.3"></a>
## [1.1.3](https://github.com/iiroj/react-style-reset/compare/v1.1.2...v1.1.3) (2019-01-13)



<a name="1.1.2"></a>
## [1.1.2](https://github.com/iiroj/react-style-reset/compare/v1.1.1...v1.1.2) (2019-01-13)



<a name="1.1.1"></a>
## [1.1.1](https://github.com/iiroj/react-style-reset/compare/v1.1.0...v1.1.1) (2019-01-09)



<a name="1.1.0"></a>
# [1.1.0](https://gitlab.com/iiroj/react-style-reset/compare/v1.0.2...v1.1.0) (2019-01-03)


### Bug Fixes

* update example typings ([9f66bb8](https://gitlab.com/iiroj/react-style-reset/commit/9f66bb8))


### Features

* build with Rollup, include cjs, esm and umd builds ([c021299](https://gitlab.com/iiroj/react-style-reset/commit/c021299))



<a name="1.0.2"></a>
## [1.0.2](https://gitlab.com/iiroj/react-style-reset/compare/v1.0.1...v1.0.2) (2018-11-30)



<a name="1.0.1"></a>
## [1.0.1](https://gitlab.com/iiroj/react-style-reset/compare/v1.0.0...v1.0.1) (2018-11-10)



<a name="1.0.0"></a>
# [1.0.0](https://gitlab.com/iiroj/react-style-reset/compare/v0.2.1...v1.0.0) (2018-10-19)


### Code Refactoring

* remove types ([f190499](https://gitlab.com/iiroj/react-style-reset/commit/f190499))


### BREAKING CHANGES

* I opt to not use emotion’s types but instead just rely on the exports being vanilla objects.



<a name="0.2.1"></a>
## [0.2.1](https://gitlab.com/iiroj/react-style-reset/compare/v0.2.0...v0.2.1) (2018-10-17)


### Bug Fixes

* use correct glob pattern to include files ([88a143e](https://gitlab.com/iiroj/react-style-reset/commit/88a143e))



<a name="0.2.0"></a>
# [0.2.0](https://gitlab.com/iiroj/react-style-reset/compare/v0.1.4...v0.2.0) (2018-10-17)


### Features

* **normalize:** add normalize.css ([2d04dc0](https://gitlab.com/iiroj/react-style-reset/commit/2d04dc0))



<a name="0.1.4"></a>
## [0.1.4](https://gitlab.com/iiroj/react-style-reset/compare/v0.1.3...v0.1.4) (2018-10-17)


### Bug Fixes

* enable prettier in tslint.json ([84e59a2](https://gitlab.com/iiroj/react-style-reset/commit/84e59a2))



<a name="0.1.3"></a>
## [0.1.3](https://gitlab.com/iiroj/react-style-reset/compare/v0.1.2...v0.1.3) (2018-10-17)



<a name="0.1.2"></a>
## [0.1.2](https://gitlab.com/iiroj/react-style-reset/compare/v0.1.1...v0.1.2) (2018-10-10)



<a name="0.1.1"></a>
## [0.1.1](https://gitlab.com/iiroj/react-style-reset/compare/v0.1.0...v0.1.1) (2018-10-10)



<a name="0.1.0"></a>
# 0.1.0 (2018-10-10)


### Features


rollup.config.js
import commonjs from "rollup-plugin-commonjs";
import resolve from "rollup-plugin-node-resolve";
import { terser } from "rollup-plugin-terser";
import typescript from "rollup-plugin-typescript2";

import pkg from "./package.json";

const production = !process.env.ROLLUP_WATCH;

const plugins = [
  typescript({ typescript: require("typescript") }),
  production && terser(),
];

const umdPlugins = [...plugins, resolve(), commonjs()];

export default [
  {
    input: "index.ts",
    output: [
      {
        exports: "named",
        file: pkg.main,
        format: "cjs",
      },
      {
        exports: "named",
        file: pkg.module,
        format: "esm",
      },
    ],
    plugins,
  },
  {
    input: "string.ts",
    output: [
      {
        exports: "named",
        file: "string.js",
        format: "cjs",
      },
      {
        exports: "named",
        file: "string.mjs",
        format: "esm",
      },
    ],
    plugins,
  },
  {
    input: "index.ts",
    output: {
      exports: "named",
      file: pkg.browser,
      format: "umd",
      name: "reactStyleReset",
    },
    plugins: umdPlugins,
  },
  {
    input: "string.ts",
    output: {
      exports: "named",
      file: "string.umd.js",
      format: "umd",
      name: "reactStyleReset",
    },
    plugins: umdPlugins,
  },
];


reset.css.object.ts
// http://meyerweb.com/eric/tools/css/reset/
// v2.0 | 20110126
// License: none (public domain)

const generics = {
  margin: 0,
  padding: 0,
  border: 0,
  fontSize: "100%",
  font: "inherit",
  verticalAlign: "baseline",
};

const displayBlock = {
  display: "block",
};

const listStyleNone = {
  listStyle: "none",
};

const contentNone = {
  content: "none",
};

const quotesNone = {
  quotes: "none",
};

const reset = {
  a: generics,
  abbr: generics,
  acronym: generics,
  address: generics,
  applet: generics,
  article: { ...generics, ...displayBlock },
  aside: { ...generics, ...displayBlock },
  audio: generics,
  b: generics,
  big: generics,
  blockquote: {
    ...generics,
    ...quotesNone,
    "&::after": contentNone,
    "&::before": contentNone,
  },
  body: { ...generics, lineHeight: 1 },
  canvas: generics,
  caption: generics,
  center: generics,
  cite: generics,
  code: generics,
  dd: generics,
  del: generics,
  details: { ...generics, ...displayBlock },
  dfn: generics,
  div: generics,
  dl: generics,
  dt: generics,
  em: generics,
  embed: generics,
  fieldset: generics,
  figcaption: { ...generics, ...displayBlock },
  figure: { ...generics, ...displayBlock },
  footer: { ...generics, ...displayBlock },
  form: generics,
  h1: generics,
  h2: generics,
  h3: generics,
  h4: generics,
  h5: generics,
  h6: generics,
  header: { ...generics, ...displayBlock },
  hgroup: { ...generics, ...displayBlock },
  html: generics,
  i: generics,
  iframe: generics,
  img: generics,
  ins: generics,
  kbd: generics,
  label: generics,
  legend: generics,
  li: generics,
  mark: generics,
  menu: { ...generics, ...displayBlock },
  nav: { ...generics, ...displayBlock },
  object: generics,
  ol: { ...generics, ...listStyleNone },
  output: generics,
  p: generics,
  pre: generics,
  q: {
    ...generics,
    ...quotesNone,
    "&::after": contentNone,
    "&::before": contentNone,
  },
  ruby: generics,
  s: generics,
  samp: generics,
  section: { ...generics, ...displayBlock },
  small: generics,
  span: generics,
  strike: generics,
  strong: generics,
  sub: generics,
  summary: generics,
  sup: generics,
  table: {
    ...generics,
    borderCollapse: "collapse" as const,
    borderSpacing: 0,
  },
  tbody: generics,
  td: generics,
  tfoot: generics,
  th: generics,
  thead: generics,
  time: generics,
  tr: generics,
  tt: generics,
  u: generics,
  ul: { ...generics, ...listStyleNone },
  var: generics,
  video: generics,
};

export default reset;


reset.css.string.ts
/* http://meyerweb.com/eric/tools/css/reset/
   v2.0 | 20110126
   License: none (public domain)
*/

const reset = `
  html, body, div, span, applet, object, iframe,
  h1, h2, h3, h4, h5, h6, p, blockquote, pre,
  a, abbr, acronym, address, big, cite, code,
  del, dfn, em, img, ins, kbd, q, s, samp,
  small, strike, strong, sub, sup, tt, var,
  b, u, i, center,
  dl, dt, dd, ol, ul, li,
  fieldset, form, label, legend,
  table, caption, tbody, tfoot, thead, tr, th, td,
  article, aside, canvas, details, embed, 
  figure, figcaption, footer, header, hgroup, 
  menu, nav, output, ruby, section, summary,
  time, mark, audio, video {
    margin: 0;
    padding: 0;
    border: 0;
    font-size: 100%;
    font: inherit;
    vertical-align: baseline;
  }
  article, aside, details, figcaption, figure, 
  footer, header, hgroup, menu, nav, section {
    display: block;
  }
  body {
    line-height: 1;
  }
  ol, ul {
    list-style: none;
  }
  blockquote, q {
    quotes: none;
  }
  blockquote:before, blockquote:after,
  q:before, q:after {
    content: '';
    content: none;
  }
  table {
    border-collapse: collapse;
    border-spacing: 0;
  }
`;

export default reset;


normalize.css.string.ts
/*! normalize.css v8.0.0 | MIT License | github.com/necolas/normalize.css */

export default `
  html {
    line-height: 1.15;
    -webkit-text-size-adjust: 100%;
  }

  body {
    margin: 0;
  }

  h1 {
    font-size: 2em;
    margin: 0.67em 0;
  }

  hr {
    box-sizing: content-box;
    height: 0;
    overflow: visible;
  }

  pre {
    font-family: monospace, monospace;
    font-size: 1em;
  }

  a {
    background-color: transparent;
  }

  abbr[title] {
    border-bottom: none;
    text-decoration: underline;
    text-decoration: underline dotted;
  }

  b,
  strong {
    font-weight: bolder;
  }

  code,
  kbd,
  samp {
    font-family: monospace, monospace;
    font-size: 1em;
  }

  small {
    font-size: 80%;
  }

  sub,
  sup {
    font-size: 75%;
    line-height: 0;
    position: relative;
    vertical-align: baseline;
  }

  sub {
    bottom: -0.25em;
  }

  sup {
    top: -0.5em;
  }

  img {
    border-style: none;
  }

  button,
  input,
  optgroup,
  select,
  textarea {
    font-family: inherit;
    font-size: 100%;
    line-height: 1.15;
    margin: 0;
  }

  button,
  input { 
    overflow: visible;
  }

  button,
  select { 
    text-transform: none;
  }

  button,
  [type="button"],
  [type="reset"],
  [type="submit"] {
    -webkit-appearance: button;
  }

  button::-moz-focus-inner,
  [type="button"]::-moz-focus-inner,
  [type="reset"]::-moz-focus-inner,
  [type="submit"]::-moz-focus-inner {
    border-style: none;
    padding: 0;
  }

  button:-moz-focusring,
  [type="button"]:-moz-focusring,
  [type="reset"]:-moz-focusring,
  [type="submit"]:-moz-focusring {
    outline: 1px dotted ButtonText;
  }

  fieldset {
    padding: 0.35em 0.75em 0.625em;
  }

  legend {
    box-sizing: border-box;
    color: inherit;
    display: table;
    max-width: 100%;
    padding: 0;
    white-space: normal;
  }

  progress {
    vertical-align: baseline;
  }

  textarea {
    overflow: auto;
  }

  [type="checkbox"],
  [type="radio"] {
    box-sizing: border-box;
    padding: 0;
  }

  [type="number"]::-webkit-inner-spin-button,
  [type="number"]::-webkit-outer-spin-button {
    height: auto;
  }

  [type="search"] {
    -webkit-appearance: textfield;
    outline-offset: -2px;
  }

  [type="search"]::-webkit-search-decoration {
    -webkit-appearance: none;
  }

  ::-webkit-file-upload-button {
    -webkit-appearance: button;
    font: inherit;
  }

  details {
    display: block;
  }

  summary {
    display: list-item;
  }

  template {
    display: none;
  }

  [hidden] {
    display: none;
  }
`;


normalize.css.object.ts
/*! normalize.css v8.0.0 | MIT License | github.com/necolas/normalize.css */

const monospace = {
  fontFamily: "monospace, monospace",
  fontSize: "1em",
};

const input = {
  fontFamily: "inherit",
  fontSize: "100%",
  lineHeight: "1.15",
  margin: "0",
};

const button = {
  WebkitAppearance: "button",
  "&::-moz-focus-inner": {
    borderStyle: "none",
    padding: "0",
  },
  "&:-moz-focusring": {
    outline: "1px dotted ButtonText",
  },
};

const checkbox = {
  boxSizing: "border-box",
  padding: "0",
};

const normalize = {
  html: {
    lineHeight: "1.15",
    WebkitTextSizeAdjust: "100%",
  },

  body: {
    margin: "0",
  },

  h1: {
    fontSize: "2em",
    margin: "0.67em 0",
  },

  hr: {
    boxSizing: "content-box",
    height: "0",
    overflow: "visible",
  },

  pre: {
    fontFamily: "monospace, monospace",
    fontSize: "1em",
  },

  a: {
    backgroundColor: "transparent",
  },

  "abbr[title]": {
    borderBottom: "none",
    // textDecoration: 'underline',
    textDecoration: "underline dotted",
  },

  strong: {
    fontWeight: "bolder",
  },
  b: {
    fontWeight: "bolder",
  },

  code: monospace,
  kbd: monospace,
  sam: monospace,

  small: {
    fontSize: "80%",
  },

  sub: {
    bottom: "-0.25em",
    fontSize: "75%",
    lineHeight: "0",
    position: "relative",
    verticalAlign: "baseline",
  },

  sup: {
    fontSize: "75%",
    lineHeight: "0",
    position: "relative",
    top: "-0.5em",
    verticalAlign: "baseline",
  },

  img: {
    borderStyle: "none",
  },

  button: {
    ...input,
    ...button,
    overflow: "visible",
    textTransform: "none",
    WebkitAppearance: "button",
  },
  input: { ...input, overflow: "visible" },
  optgroup: input,
  select: { ...input, textTransform: "none" },
  textare: input,

  '[type="button"]': button,
  '[type="reset"]': button,
  '[type="submit"]': button,

  fieldset: {
    padding: "0.35em 0.75em 0.625em",
  },

  legend: {
    boxSizing: "border-box",
    color: "inherit",
    display: "table",
    maxWidth: "100%",
    padding: "0",
    whiteSpace: "normal",
  },

  progress: {
    verticalAlign: "baseline",
  },

  textarea: {
    overflow: "auto",
  },

  '[type="checkbox"]': checkbox,
  '[type="radio"]': checkbox,

  '[type="number"]::-webkit-inner-spin-button': {
    height: "auto",
  },
  '[type="number"]::-webkit-outer-spin-button': {
    height: "auto",
  },

  '[type="search"]': {
    WebkitAppearance: "textfield",
    outlineOffset: "-2px",
  },

  '[type="search"]::-webkit-search-decoration': {
    WebkitAppearance: '""',
  },

  "::-webkit-file-upload-button": {
    WebkitAppearance: "button",
    font: "inherit",
  },

  details: {
    display: "block",
  },

  summary: {
    display: "list-item",
  },

  template: {
    display: "none",
  },

  "[hidden]": {
    display: "none",
  },
};

export default normalize;
